test/plots/matrix-diagram.js

```diff
@@ -0,0 +1,50 @@
+import * as Plot from "@observablehq/plot";
+import * as d3 from "d3";
+
+export default async function () {
+  const data = [
+    {x: "b", y: "a", value: 2},
+    {x: "c", y: "b", value: 1},
+    {x: "d", y: "a", value: 2},
+    {x: "d", y: "b", value: 2}
+  ];
+  const domain = [
+    ...new Set([...Plot.valueof(data, "x"), ...Plot.valueof(data, "y")])
+  ].sort();
```

Comment:
For your consideration…

```suggestion
  const domain = d3.sort(d3.union(d3.map(data, d => d.x), d3.map(data, d => d.y)));
```